common/src/main/java/com/revenuecat/purchases/common/Config.kt

```diff
@@ -2,7 +2,7 @@ package com.revenuecat.purchases.common
 
 object Config {
 
-    var debugLogsEnabled = false
+    var debugLogsEnabled = BuildConfig.DEBUG
```

Comment:
Have you successfully tested this? I don't think libraries have access to an app's BuildConfig 🤔

Reply:
I believe we need something similar to this https://stackoverflow.com/a/37941079/1148405

Reply:
I hadn't tested it, sorry!
I'm obviously not super familiar with this, but wouldn't `BuildConfig.DEBUG` be `true` or `false` depending on how the library is built?

Reply:
We distribute the release variant of the library, which means that the `BuildConfig.DEBUG` you're accessing will always be `false`. Here's an example while building MagicWeather for debug:

BuildConfig.java in `app/build/google/debug/com.revenuecat.sample/`
```java
/**
 * Automatically generated file. DO NOT MODIFY
 */
package com.revenuecat.sample;

public final class BuildConfig {
  public static final boolean DEBUG = Boolean.parseBoolean("true");
  public static final String APPLICATION_ID = "com.revenuecat.purchases_sample";
  public static final String BUILD_TYPE = "debug";
  public static final String FLAVOR = "google";
  public static final int VERSION_CODE = 1;
  public static final String VERSION_NAME = "1.0";
  // Field from product flavor: google
  public static final String STORE = "google";
}
```

BuildConfig.class inside the downloaded aar 
<img width="546" alt="Screen Shot 2022-02-15 at 8 29 20 AM" src="https://user-images.githubusercontent.com/664544/154105351-298f20c2-46e6-45bf-ae4a-7a54ac71824f.png">

```java
package com.revenuecat.purchases;

public final class BuildConfig {
    public static final boolean DEBUG = false;
    public static final String LIBRARY_PACKAGE_NAME = "com.revenuecat.purchases";
    public static final String BUILD_TYPE = "release";

    public BuildConfig() {
    }
}
```

What they suggest in https://stackoverflow.com/a/37941079/1148405 is to access the BuildConfig of the app, which will be the one that actually changes when building for debug and for release.

Reply:
> We distribute the release variant of the library, which means that the BuildConfig.DEBUG you're accessing will always be false.

That's the plan, for this to _default_ to true / false depending on the type of build. That's what https://github.com/RevenueCat/purchases-ios/pull/1262 does too.
In production, this doesn't change anything, it still defaults to `false`, but locally or when running tests we'll now see debug logs. In iOS depending on the type of integration, if the framework is pre-built, it will also be release, there's no way to determine this at runtime.

Does that make sense? Again, this would just be the new default, but clients can still change it.

Reply:
How do clients change it though? I guess that's my question. If this is more for us during tests or for devs not using Maven, it looks good!

Reply:
Clients can still change `debugLogsEnabled` like before, this only changes the default value to a better default. I agree the runtime check might make more sense in some instances but might not be worth it cause we'd have to add code to check that the users haven't already changed the value.

Reply:
@vegaro does that make sense?

Reply:
Sorry I missed your reply. I see what you meant with clients can still change it. Makes sense then. I will approve!